vars/filebeat.groovy

```diff
@@ -100,8 +121,9 @@ def stop(Map args = [:]){
     docker exec -t ${stepConfig.id} chmod -R ugo+rw /output || echo "Exit code \$?"
     docker stop --time ${timeout} ${stepConfig.id} || echo "Exit code \$?"
   """)
-  if(archiveOnlyOnFail == false || (archiveOnlyOnFail && isBuildFailure())){
-    archiveArtifacts(artifacts: "**/${stepConfig.output}*", allowEmptyArchive: true)
+
+  if(!archiveOnlyOnFail || isBuildFailure()){
+    archiveArtifacts(artifacts: "./${stepConfig.output}*", allowEmptyArchive: true)
```

Comment:
This example will not archive the logs with the new code

```
filebeat(wordir: 'logs'){
...
}
```

Reply:
Is this unit test https://github.com/elastic/apm-pipeline-library/pull/1027/files#diff-4ec712ed1223a6369b2697f148d91657ea6dc9216f60bd739588c6f85a39fd17R270 covering that use case? I see it archiving the artifacts

Reply:
```
filebeat(wordir: 'logs', output: 'foo.log'){ 
 echo 'hello'
}
```

this will create `${WORKSPACE}/logs/foo.log` and will try to archive  `archiveArtifacts(artifacts: "./foo.log*", allowEmptyArchive: true)` that it is `${WORKSPACE}/./foo.log*` does not match so it will archive nothing.

Reply:
Will take a look at this, thanks!